parquet-hadoop/src/test/java/org/apache/parquet/encodings/TestTypeEncodings.java

```diff
@@ -0,0 +1,417 @@
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.io.api.PrimitiveConverter;
+import org.apache.parquet.schema.*;
+import org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName;
+import org.apache.parquet.statistics.RandomValues;
+import org.apache.parquet.statistics.TestStatistics;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+import static junit.framework.Assert.assertEquals;
+import static org.apache.parquet.column.ParquetProperties.WriterVersion.PARQUET_1_0;
+import static org.apache.parquet.column.ParquetProperties.WriterVersion.PARQUET_2_0;
+
+@RunWith(Parameterized.class)
+public class TestTypeEncodings {
+  private static final int RANDOM_SEED = 1;
+  private static final int RECORD_COUNT = 1000000;
+  private static final int FIXED_LENGTH = 60;
+
+  private static final Configuration configuration = new Configuration();
+
+  private static RandomValues.IntGenerator intGenerator;
+  private static RandomValues.LongGenerator longGenerator;
+  private static RandomValues.Int96Generator int96Generator;
+  private static RandomValues.FloatGenerator floatGenerator;
+  private static RandomValues.DoubleGenerator doubleGenerator;
+  private static RandomValues.BinaryGenerator binaryGenerator;
+  private static RandomValues.FixedGenerator fixedBinaryGenerator;
+
+  // Parameters
+  private PrimitiveTypeName paramTypeName;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> getParameters() {
+    return Arrays.asList(new Object[][] {
+        { PrimitiveTypeName.BOOLEAN },
+        { PrimitiveTypeName.INT32 },
+        { PrimitiveTypeName.INT64 },
+        { PrimitiveTypeName.INT96 },
+        { PrimitiveTypeName.FLOAT },
+        { PrimitiveTypeName.DOUBLE },
+        { PrimitiveTypeName.BINARY },
+        { PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY }
+    });
+  }
+
+  public TestTypeEncodings(PrimitiveTypeName typeName) {
+    this.paramTypeName = typeName;
+  }
+
+  @BeforeClass
+  public static void initialize() throws IOException {
+    Random random = new Random(RANDOM_SEED);
+    intGenerator = new RandomValues.IntGenerator(random.nextLong());
+    longGenerator = new RandomValues.LongGenerator(random.nextLong());
+    int96Generator = new RandomValues.Int96Generator(random.nextLong());
+    floatGenerator = new RandomValues.FloatGenerator(random.nextLong());
+    doubleGenerator = new RandomValues.DoubleGenerator(random.nextLong());
+    binaryGenerator = new RandomValues.BinaryGenerator(random.nextLong());
+    fixedBinaryGenerator = new RandomValues.FixedGenerator(random.nextLong(), FIXED_LENGTH);
+  }
+
+  @Test
+  public void testTypeEncodings() throws Exception {
+    int pageSize = 64; // 64B
```

Comment:
I think we should use a bigger value for this. I bumped it up to 16k and the tests work just fine. 8k or 16k is a setting more like what we would see in real data, and you can still fit several pages in each row group.